sky/templates/scp-ray.yml.j2

```diff
@@ -58,49 +58,48 @@ initialization_commands: []
 #   current num items (num SSH connections): 1
 setup_commands:
   # Disable `unattended-upgrades` to prevent apt-get from hanging. It should be called at the beginning before the process started to avoid being blocked. (This is a temporary fix.)
-  # Create ~/.ssh/config file in case the file does not exist in the image.
-  # Line 'rm ..': there is another installation of pip.
+  # Create ~/.ssh/config file in case the file does not exist in the custom image.
+  # Make sure python3 & pip3 are available on this image.
+  # We set auto_activate_base to be false for pre-installed conda.
+  # This also kills the service that is holding the lock on dpkg (problem only exists on aws/azure, not gcp)
   # Line 'sudo bash ..': set the ulimit as suggested by ray docs for performance. https://docs.ray.io/en/latest/cluster/vms/user-guides/large-cluster-best-practices.html#system-configuration
   # Line 'sudo grep ..': set the number of threads per process to unlimited to avoid ray job submit stucking issue when the number of running ray jobs increase.
   # Line 'mkdir -p ..': disable host key check
   # Line 'python3 -c ..': patch the buggy ray files and enable `-o allow_other` option for `goofys`
-  - sudo systemctl stop unattended-upgrades || true;
-    sudo systemctl disable unattended-upgrades || true;
-    sudo sed -i 's/Unattended-Upgrade "1"/Unattended-Upgrade "0"/g' /etc/apt/apt.conf.d/20auto-upgrades || true;
-    sudo kill -9 `sudo lsof /var/lib/dpkg/lock-frontend | awk '{print $2}' | tail -n 1` || true;
-    sudo pkill -9 apt-get;
-    sudo pkill -9 dpkg;
-    sudo dpkg --configure -a;
-    mkdir -p ~/.ssh; touch ~/.ssh/config;
-    rm ~/.local/bin/pip ~/.local/bin/pip3 ~/.local/bin/pip3.8 ~/.local/bin/pip3.10;
```

Comment:
Do we want to still keep these lines? It stops the unattended upgrade for apt, which can hold the lock for the user trying `sudo apt install`. We need to confirm whether the unattended upgrade is enabled in the image we are using.
We don’t include this in the `aws-ray.yaml` because we are using the cloud-init in the following place to disable the unattended upgrade.
https://github.com/skypilot-org/skypilot/blob/88c7a9e277171aaa6ba5086abce52b0c4e6f421f/sky/templates/aws-ray.yml.j2#L78-L87